src/deadline/job_attachments/asset_sync.py

```diff
@@ -68,6 +69,26 @@
 logger = getLogger("deadline.job_attachments")
 
 
+@dataclass
+class PathMappingRule:
+    source_path_format: str
+    """The path format associated with the source path (windows vs posix)"""
+
+    source_path: str
+    """The path we're looking to change"""
+
+    destination_path: str
+    """The path to transform the source path to"""
+
+    def __init__(self, source_path_format: str, source_path: str, destination_path: str):
+        self.source_path_format = source_path_format
+        self.source_path = source_path
+        self.destination_path = destination_path
+
+    def to_dict(self) -> dict[str, str]:
+        return {k: str(v) for k, v in asdict(self).items()}
```

Comment:
How come `asdict()` not good enough? is it just for the string cast?

Reply:
Good catch, `asdict()` should be good enough. I can remove it.